In [1]:
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

# 데이터 정리
### 영화정보
> meta19 meta20 meta21

### 리뷰
> 네이버 : n19 | n20 | n21<br>
> 다음 : d19 | d20 | d21<br>
> 왓챠피디아 : w19 | w20 | w21

# 데이터 병합

In [2]:
meta = pd.read_csv('data/review/meta192021.csv')

In [3]:
w19 = pd.read_csv('data/review/Wa_review2019_last.csv')

In [4]:
w20 = pd.read_csv('data/review/Wa_review2020_last.csv')

In [5]:
w21 = pd.read_csv('data/review/Wa_review2021_last.csv')

In [6]:
w1920 = pd.merge(w19, w20, on=('title','score','content','write_date','source_site'), how='outer')

In [7]:
w192021 = pd.merge(w1920, w21, on=('title','score','content','write_date','source_site'), how='outer')

In [8]:
w192021 = w192021.reset_index(drop=True)

In [9]:
w192021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6804 entries, 0 to 6803
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   title        6804 non-null   object 
 1   score        6804 non-null   object 
 2   content      6570 non-null   object 
 3   write_date   0 non-null      float64
 4   source_site  6804 non-null   object 
dtypes: float64(1), object(4)
memory usage: 265.9+ KB


# 전처리
* 중복값 제거 (6,804 -> 6,754 -> 6,569)
* `페르소나`, `각오는 됐나, 거기 여자`는 네이버에서 2018 개봉이라 제외
* `힐빌리의 노래`, `러브라인`, `황무지 5월의 고해`, `오버 더 문`, `바이올렛 에버가든 외전 -영원과 자동수기인형-` 등은 네이버 영화 정보에 없음
* `웰컴투X-월드`, `버즈 오브 프레이(할리 퀸의 황홀한 해방)`, `작은빛`은 띄어쓰기가 달라서 네이버로 통일

In [10]:
no_duplicates = w192021.drop_duplicates()
no_duplicates = no_duplicates[no_duplicates['content'].isnull()==False]
no_duplicates = no_duplicates.reset_index(drop=True)

In [11]:
def enc_title(x):
    if x == "웰컴투X-월드":
        return '웰컴 투 X-월드'
    
    elif x == "버즈 오브 프레이: 할리 퀸의 황홀한 해방":
        return '버즈 오브 프레이(할리 퀸의 황홀한 해방)'
    
    elif x == "작은빛":
        return "작은 빛"
    
    else: #일치 하지 않으면 input을 다시 return해줘야 함. 그렇지 않으면 null로 채워짐(아마도)
        return x
no_duplicates['title'] = no_duplicates['title'].apply(enc_title)

In [12]:
cnt = 0
titles = meta['title'].values
for i in range(len(no_duplicates)):
    title = no_duplicates.iloc[i,0]
    content = no_duplicates.iloc[i,2]
    if title not in titles:
        cnt+=1
        print(i, title)
print(cnt)

605 페르소나
606 페르소나
607 페르소나
608 페르소나
609 페르소나
610 페르소나
611 페르소나
612 페르소나
613 페르소나
614 페르소나
615 페르소나
616 페르소나
617 페르소나
618 페르소나
619 페르소나
620 페르소나
621 페르소나
622 페르소나
623 페르소나
645 각오는 됐나, 거기 여자
646 각오는 됐나, 거기 여자
647 각오는 됐나, 거기 여자
648 각오는 됐나, 거기 여자
649 각오는 됐나, 거기 여자
650 각오는 됐나, 거기 여자
651 각오는 됐나, 거기 여자
652 각오는 됐나, 거기 여자
653 각오는 됐나, 거기 여자
654 각오는 됐나, 거기 여자
655 각오는 됐나, 거기 여자
656 각오는 됐나, 거기 여자
657 각오는 됐나, 거기 여자
658 각오는 됐나, 거기 여자
659 각오는 됐나, 거기 여자
660 각오는 됐나, 거기 여자
661 각오는 됐나, 거기 여자
662 각오는 됐나, 거기 여자
663 각오는 됐나, 거기 여자
664 각오는 됐나, 거기 여자
665 각오는 됐나, 거기 여자
1256 힐빌리의 노래
1257 힐빌리의 노래
1258 힐빌리의 노래
1304 러브라인
1305 러브라인
1306 러브라인
1352 황무지 5월의 고해
1353 황무지 5월의 고해
1354 황무지 5월의 고해
1355 황무지 5월의 고해
1356 황무지 5월의 고해
1357 황무지 5월의 고해
1358 황무지 5월의 고해
1359 황무지 5월의 고해
1360 황무지 5월의 고해
1361 황무지 5월의 고해
1435 오버 더 문
1436 오버 더 문
1437 오버 더 문
1438 오버 더 문
1439 오버 더 문
1440 오버 더 문
1441 오버 더 문
1442 오버 더 문
1443 오버 더 문
1444 오버 더 문
1445 오버 더 문
1446 오버 더 문
1447 오버 더 문
1448 오버 더 문
1449 오버 더 문
1450 오버 더 문
1451 오버 더 문
1452 오버 더 문
1453 오

In [13]:
meta[meta['title'].str.contains('코난')==True]

title        genre nation running_time    release_date  \
236      명탐정 코난: 감청의 권        애니메이션     일본         109분  2019 .07.24 개봉   
237      명탐정 코난:전율의 악보  애니메이션, 서스펜스     일본         116분  2019 .02.14 개봉   
1545  명탐정 코난: 진홍의 수학여행        애니메이션     일본          92분  2021 .01.27 개봉   

        director                                              actor    grade  \
236      나가오카 치카                        타카야마 미나미, 야마자키 와카나, 코야마 리키야  12세 관람가   
237   야마모토 야스이치로  타카야마 미나미(에도가와 코난), 야마자키 와카나(모리 란), 쿠와시마 호우코(아키...  12세 관람가   
1545  야마모토 야스이치로                        야마구치 캇페이, 야마자키 와카나, 히다카 노리코  12세 관람가   

                                                summary  \
236   19세기말 해적선과 함께 싱가포르 인근 바다에 가라앉았다고 전해지는\n 전설의 보물...   
237   음악가만을 노린 연쇄살인사건이 발생하고,\n 피해자들은 모두 유명한 피아니스트가 설...   
1545  초등학생 코난이 아닌, 본래의 모습으로 교토 수학여행에 참가하게 된 고등학생 탐정 ...   

                                                 poster  
236   https://movie-phinf.pstatic.net/20190626_238/1...  
237   https://movie-phinf.pstatic.net/20190104_172/1...  
1545  https://movie-phinf.pstatic.net/20210113_148/1...

In [14]:
no_duplicates.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6569 entries, 0 to 6568
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   title        6569 non-null   object 
 1   score        6569 non-null   object 
 2   content      6569 non-null   object 
 3   write_date   0 non-null      float64
 4   source_site  6569 non-null   object 
dtypes: float64(1), object(4)
memory usage: 256.7+ KB


#### csv 저장

In [15]:
no_duplicates.to_csv('./data/review/Wa_review_192021.csv',index=False)

In [16]:
movie_all = pd.merge(meta, no_duplicates
    , how = 'inner'
    , on = ("title"))

In [17]:
movie_all.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6228 entries, 0 to 6227
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   title         6228 non-null   object 
 1   genre         6228 non-null   object 
 2   nation        6228 non-null   object 
 3   running_time  6228 non-null   object 
 4   release_date  6228 non-null   object 
 5   director      6228 non-null   object 
 6   actor         6228 non-null   object 
 7   grade         6228 non-null   object 
 8   summary       6228 non-null   object 
 9   poster        6228 non-null   object 
 10  score         6228 non-null   object 
 11  content       6228 non-null   object 
 12  write_date    0 non-null      float64
 13  source_site   6228 non-null   object 
dtypes: float64(1), object(13)
memory usage: 729.8+ KB


In [18]:
mkdir data/total

In [19]:
ls

1013_naver_movie_sim.ipynb  add_movie.py    add_score.py  tag_search.py
Daum_merge.ipynb            add_ranking.py  data/
Watcha_merge.ipynb          add_review.py   scraping.py


In [20]:
movie_all.to_csv('data/total/Wa_192021.csv',index=False)